### PR TITLE
[DF-376] Tracking when a user sorts results or changes result view

### PR DESCRIPTION
**Pull request by janehughes**
Ticket URL: https://national-archives.atlassian.net/browse/DF-376

## About these changes

Adds JS to push current active 'sort by' option to the data layer. This also updates the existing js to update the active value when the user selects a new sort by option and then sends it to GTM on form submission.

## How to check these changes

Run a test 'sort by' filter locally (check the dataLayer in devtools console and see the 'sort-results' event occur after form submission.

GTM should _only_ receive an event when the user changes the select value and submits the form.

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [ ] Added/updated tests and documentation where relevant. n/a

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: templates/search/blocks/search_filters.html
Comment (ababic) [inline templates/search/blocks/search_filters.html:1]: We should really generate this value in Python and include it in the context. The various views can then override where the value comes from.

Comment (janehughes) [reply templates/search/blocks/search_filters.html:1]: That would be ideal if possible @ababic

File: templates/search/blocks/search_sort_and_view_options.html [hunk 1]
```diff
@@ -2,12 +2,12 @@
 
 
 <div class="search-sort-view">
-    <form method="GET" class="search-sort-view__form search-sort-view__desktop" data-id="sort-form" id="catalogue-sort-form-desktop" search-type="{% with request.resolver_match.url_name as url_name  %} {% if url_name == 'search-featured' %} {{searchtabs.ALL.value}} {% elif url_name == 'search-catalogue' %}{{searchtabs.CATALOGUE.value}}{% elif url_name == 'search-website' %}
-{{searchtabs.WEBSITE.value}} {% endif %} {%endwith%}"  search-bucket="{{buckets.current.label}}" search-filter-name="Sort results" search-filter-value="">
+    <form method="GET" class="search-sort-view__form search-sort-view__desktop" data-id="sort-form" id="catalogue-sort-form-desktop" data-search-type="{% with request.resolver_match.url_name as url_name %}{% if url_name == 'search-featured' %}{{ searchtabs.ALL.value }}{% elif url_name == 'search-catalogue' %}{{ searchtabs.CATALOGUE.value }}{% elif url_name == 'search-website' %}{{ searchtabs.WEBSITE.value }}{% endif %}{% endwith %}" data-search-bucket="{{ buckets.current.label }}" data-search-filter-name="Sort results" data-search-filter-value="">
```
Comment (ababic) [inline templates/search/blocks/search_sort_and_view_options.html:5]:
```suggestion
    <form method="GET" class="search-sort-view__form search-sort-view__desktop" data-id="sort-form" id="catalogue-sort-form-desktop" data-search-type="{{ view.search_tab }}" data-search-bucket="{{ buckets.current.label }}" data-search-filter-name="Sort results" data-search-filter-value="">
```

**Review by ababic — CHANGES_REQUESTED** [review]
@janehughes Looks like this value was already being set for datalayer generation, so it should 
 already be possible to access it from the view instance directly

File: templates/search/blocks/search_sort_and_view_options.html [hunk 2]
```diff
@@ -26,7 +26,7 @@ <h2 class="search-sort-view__heading"><span class="sr-only">Display as:</span></
...
                 </span>
             {% else %}
-                <a href="?{% query_string_include 'display' 'list' %}" id="listview" search-type="{% with request.resolver_match.url_name as url_name  %} {% if url_name == 'search-featured' %} {{searchtabs.ALL.value}} {% elif url_name == 'search-catalogue' %} {{searchtabs.CATALOGUE.value}} {% elif url_name == 'search-website' %}{{searchtabs.WEBSITE.value}}{% endif %} {%endwith%}"  search-bucket="{{buckets.current.label}}" search-filter-name="Sort results" search-filter-value="Display results as a list" class="search-sort-view__icon">
+                <a href="?{% query_string_include 'display' 'list' %}" id="listview" search-type="{% with request.resolver_match.url_name as url_name %}{% if url_name == 'search-featured' %}{{ searchtabs.ALL.value }}{% elif url_name == 'search-catalogue' %}{{ searchtabs.CATALOGUE.value }}{% elif url_name == 'search-website' %}{{ searchtabs.WEBSITE.value }}{% endif %}{% endwith %}" search-bucket="{{ buckets.current.label }}" search-filter-name="Sort results" search-filter-value="Display results as a list" class="search-sort-view__icon">
```
Comment (ababic) [inline templates/search/blocks/search_sort_and_view_options.html:29]:
```suggestion
                <a href="?{% query_string_include 'display' 'list' %}" id="listview" search-type="{{ view.search_tab }}" search-bucket="{{ buckets.current.label }}" search-filter-name="Sort results" search-filter-value="Display results as a list" class="search-sort-view__icon">
```

File: templates/search/blocks/search_sort_and_view_options.html [hunk 3]
```diff
@@ -39,8 +39,7 @@ <h2 class="search-sort-view__heading"><span class="sr-only">Display as:</span></
...
                 </span>
             {% else %}
-                <a href="?{% query_string_include 'display' 'grid' %}" id="gridView" search-type="{% with request.resolver_match.url_name as url_name  %} {% if url_name == 'search-featured' %} {{searchtabs.ALL.value}} {% elif url_name == 'search-catalogue' %}{{searchtabs.CATALOGUE.value}}{% elif url_name == 'search-website' %}
-{{searchtabs.WEBSITE.value}} {% endif %} {%endwith%}"  search-bucket="{{buckets.current.label}}" search-filter-name="Sort results" search-filter-value="Display results as a grid" class="search-sort-view__icon">
+                <a href="?{% query_string_include 'display' 'grid' %}" id="gridView" search-type="{% with request.resolver_match.url_name as url_name %}{% if url_name == 'search-featured' %}{{ searchtabs.ALL.value }}{% elif url_name == 'search-catalogue' %}{{ searchtabs.CATALOGUE.value }}{% elif url_name == 'search-website' %}{{ searchtabs.WEBSITE.value }}{% endif %}{% endwith %}" search-bucket="{{ buckets.current.label }}" search-filter-name="Sort results" search-filter-value="Display results as a grid" class="search-sort-view__icon">
```
Comment (ababic) [inline templates/search/blocks/search_sort_and_view_options.html:42]:
```suggestion
                <a href="?{% query_string_include 'display' 'grid' %}" id="gridView" search-type="{{ view.search_tab }}" search-bucket="{{ buckets.current.label }}" search-filter-name="Sort results" search-filter-value="Display results as a grid" class="search-sort-view__icon">
```

**Review by ababic — APPROVED** [review]
If the right values are displaying in the right places, this is all good from my perspective :)

**Pull request merged**